### PR TITLE
Set the input amount max on blur

### PR DESCRIPTION
**Pull request by Tristyn**

### CODE REVIEW

File: app/src/components/YourInfo.tsx [hunk 1]
```diff
@@ -148,23 +148,30 @@ export const YourInfo = () => {
...
+  const preFillWithStakedJet = () => {
+    setInputAmount(stakedJetTokens);
+  };
+  const setInputAmountInRange = () => {
+    const maxInput = Math.max(jetBalanceTokens, stakedJetTokens);
+    if (inputAmount) {
+      const withinRange = Math.min(inputAmount, maxInput);
+      setInputAmount(withinRange);
+    }
   };
```
Comment (Tristyn) [inline app/src/components/YourInfo.tsx:175]: this

File: app/src/components/YourInfo.tsx [hunk 2]
```diff
@@ -258,16 +262,17 @@ export const YourInfo = () => {
...
-                  setInputAmount(value);
+              token
+              onChange={(value: string) => {
+                let number = Number(value);
+                if (isNaN(number) || number < 0) {
+                  number = 0;
                 }
+                setInputAmount(number);
               }}
+              onBlur={setInputAmountInRange}
```
Comment (Tristyn) [inline app/src/components/YourInfo.tsx:275]: this

File: app/src/components/YourInfo.tsx [hunk 3]
```diff
@@ -148,23 +148,30 @@ export const YourInfo = () => {
...
-    setInputAmount(sharesToTokenValueDisplay);
+  const preFillWithStakedJet = () => {
+    setInputAmount(stakedJetTokens);
+  };
+  const setInputAmountInRange = () => {
+    const maxInput = Math.max(jetBalanceTokens, stakedJetTokens);
+    if (inputAmount) {
+      const withinRange = Math.min(inputAmount, maxInput);
+      setInputAmount(withinRange);
+    }
```
Comment (Tristyn) [inline app/src/components/YourInfo.tsx:174]:
```suggestion
    if (inputAmount) {
      const maxInput = Math.max(jetBalanceTokens, stakedJetTokens);
      const withinRange = Math.min(inputAmount, maxInput);
      setInputAmount(withinRange);
    }
```

**Comment by jetjake** [comment]
@Tristyn, what are we trying to solve with this guy?

**Comment by Tristyn** [comment]
It puts the input amount in the correct range

Let's say you have 100 JET in your wallet and 0 staked
You enter 1,000,000 in the input, it changes to 100 when you switch focus cause that's the max
Works with unstaking too

**Review by jetjake — APPROVED** [review]

**Pull request merged**